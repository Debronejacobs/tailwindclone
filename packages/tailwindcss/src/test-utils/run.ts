import { optimizeCss } from '@tailwindcss/optimize'
import { compile } from '..'

export function compileCss(css: string, candidates: string[] = []) {
  return optimizeCss(compile(css).build(candidates)).trim()
}

export function run(candidates: string[]) {
  return optimizeCss(compile('@tailwind utilities;').build(candidates)).trim()
}
