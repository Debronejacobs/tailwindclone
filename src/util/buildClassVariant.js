import escapeClassName from './escapeClassName'

export default function buildClassVariant(className, variantName, variantsAsSuffix, separator) {
  if (variantsAsSuffix) {
    return `${className}${escapeClassName(separator)}${variantName}`
  }
  return `.${variantName}${escapeClassName(separator)}${className.slice(1)}`
}
