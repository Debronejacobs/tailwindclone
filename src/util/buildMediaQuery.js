import _ from 'lodash'

function isInvalidValue(value) {
  return !value || value === '' || value === 0 || value === '0'
}

export default function buildMediaQuery(screens) {
  if (isInvalidValue(screens)) {
    return undefined
  }

  if (_.isString(screens)) {
    screens = { min: screens }
  }

  if (!Array.isArray(screens)) {
    screens = [screens]
  }

  return _(screens)
    .map((screen) => {
      if (_.has(screen, 'raw')) {
        return screen.raw
      }

      return _(screen)
        .map((value, feature) => {
          feature = _.get(
            {
              min: 'min-width',
              max: 'max-width',
            },
            feature,
            feature
          )

          if (isInvalidValue(value)) {
            return undefined
          }

          return `(${feature}: ${value})`
        })
        .uniq()
        .filter((v) => v !== undefined)
        .join(' and ')
    })
    .join(', ')
}
