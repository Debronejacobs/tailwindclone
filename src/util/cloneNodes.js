import _ from 'lodash'

export default nodes => _.map(nodes, node => node.clone())
