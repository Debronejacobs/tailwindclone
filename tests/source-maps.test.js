import postcss from 'postcss'
import { parseSourceMaps } from './util/source-maps'
import { runWithSourceMaps as run, html, css, map } from './util/run'

test('apply generates source maps', async () => {
  let config = {
    content: [
      {
        raw: html`
          <div class="with-declaration"></div>
          <div class="with-comment"></div>
          <div class="just-apply"></div>
        `,
      },
    ],
    corePlugins: { preflight: false },
  }

  let input = css`
    .with-declaration {
      background-color: red;
      @apply h-4 w-4 bg-green-500;
    }

    .with-comment {
      /* sourcemap will work here too */
      @apply h-4 w-4 bg-red-500;
    }

    .just-apply {
      @apply h-4 w-4 bg-black;
    }
  `

  let result = await run(input, config)
  let { sources, annotations } = parseSourceMaps(result)

  // All CSS generated by Tailwind CSS should be annotated with source maps
  // And always be able to point to the original source file
  expect(sources).not.toContain('<no source>')
  expect(sources.length).toBe(1)

  expect(annotations).toEqual([
    '2:4 -> 2:4',
    '3:6-27 -> 3:6-27',
    '4:6-33 -> 4:6-18',
    '4:6-33 -> 5:6-17',
    '4:6-33 -> 6:6-24',
    '4:6-33 -> 7:6-61',
    '5:4 -> 8:4',
    '7:4 -> 10:4',
    '8:6-39 -> 11:6-39',
    '9:6-31 -> 12:6-18',
    '9:6-31 -> 13:6-17',
    '9:6-31 -> 14:6-24',
    '9:6-31 -> 15:6-61',
    '10:4 -> 16:4',
    '13:6 -> 18:4',
    '13:6-29 -> 19:6-18',
    '13:6-29 -> 20:6-17',
    '13:6-29 -> 21:6-24',
    '13:6 -> 22:6',
    '13:29 -> 23:0',
  ])
})

test('preflight + base have source maps', async () => {
  let config = {
    content: [],
  }

  let input = css`
    @tailwind base;
  `

  let result = await run(input, config)
  let { sources, annotations } = parseSourceMaps(result)

  // All CSS generated by Tailwind CSS should be annotated with source maps
  // And always be able to point to the original source file
  expect(sources).not.toContain('<no source>')
  expect(sources.length).toBe(1)

  expect(annotations).toEqual([
    '2:4 -> 1:0',
    '2:18-4 -> 3:1-2',
    '2:18 -> 6:1',
    '2:4 -> 8:0',
    '2:4-18 -> 11:2-32',
    '2:4-18 -> 12:2-25',
    '2:4-18 -> 13:2-29',
    '2:4-18 -> 14:2-31',
    '2:18 -> 15:0',
    '2:4 -> 17:0',
    '2:4-18 -> 19:2-18',
    '2:18 -> 20:0',
    '2:4 -> 22:0',
    '2:18 -> 30:1',
    '2:4 -> 32:0',
    '2:4-18 -> 34:2-26',
    '2:4-18 -> 35:2-40',
    '2:4-18 -> 36:2-26',
    '2:4-18 -> 37:2-21',
    '2:4-18 -> 38:2-137',
    '2:4-18 -> 39:2-39',
    '2:4-18 -> 40:2-41',
    '2:4-18 -> 41:2-50',
    '2:18 -> 42:0',
    '2:4 -> 44:0',
    '2:18 -> 47:1',
    '2:4 -> 49:0',
    '2:4-18 -> 50:2-19',
    '2:4-18 -> 51:2-30',
    '2:18 -> 52:0',
    '2:4 -> 54:0',
    '2:18 -> 58:1',
    '2:4 -> 60:0',
    '2:4-18 -> 61:2-19',
    '2:4-18 -> 62:2-24',
    '2:4-18 -> 63:2-31',
    '2:18 -> 64:0',
    '2:4 -> 66:0',
    '2:18 -> 68:1',
    '2:4 -> 70:0',
    '2:4-18 -> 71:2-35',
    '2:18 -> 72:0',
    '2:4 -> 74:0',
    '2:18 -> 76:1',
    '2:4 -> 78:0',
    '2:4-18 -> 84:2-20',
    '2:4-18 -> 85:2-22',
    '2:18 -> 86:0',
    '2:4 -> 88:0',
    '2:18 -> 90:1',
    '2:4 -> 92:0',
    '2:4-18 -> 93:2-16',
    '2:4-18 -> 94:2-26',
    '2:18 -> 95:0',
    '2:4 -> 97:0',
    '2:18 -> 99:1',
    '2:4 -> 101:0',
    '2:4-18 -> 103:2-21',
    '2:18 -> 104:0',
    '2:4 -> 106:0',
    '2:18 -> 111:1',
    '2:4 -> 113:0',
    '2:4-18 -> 117:2-121',
    '2:4-18 -> 118:2-39',
    '2:4-18 -> 119:2-41',
    '2:4-18 -> 120:2-24',
    '2:18 -> 121:0',
    '2:4 -> 123:0',
    '2:18 -> 125:1',
    '2:4 -> 127:0',
    '2:4-18 -> 128:2-16',
    '2:18 -> 129:0',
    '2:4 -> 131:0',
    '2:18 -> 133:1',
    '2:4 -> 135:0',
    '2:4-18 -> 137:2-16',
    '2:4-18 -> 138:2-16',
    '2:4-18 -> 139:2-20',
    '2:4-18 -> 140:2-26',
    '2:18 -> 141:0',
    '2:4 -> 143:0',
    '2:4-18 -> 144:2-17',
    '2:18 -> 145:0',
    '2:4 -> 147:0',
    '2:4-18 -> 148:2-13',
    '2:18 -> 149:0',
    '2:4 -> 151:0',
    '2:18 -> 155:1',
    '2:4 -> 157:0',
    '2:4-18 -> 158:2-24',
    '2:4-18 -> 159:2-31',
    '2:4-18 -> 160:2-35',
    '2:18 -> 161:0',
    '2:4 -> 163:0',
    '2:18 -> 167:1',
    '2:4 -> 169:0',
    '2:4-18 -> 174:2-30',
    '2:4-18 -> 175:2-40',
    '2:4-18 -> 176:2-42',
    '2:4-18 -> 177:2-25',
    '2:4-18 -> 178:2-30',
    '2:4-18 -> 179:2-30',
    '2:4-18 -> 180:2-24',
    '2:4-18 -> 181:2-19',
    '2:4-18 -> 182:2-20',
    '2:18 -> 183:0',
    '2:4 -> 185:0',
    '2:18 -> 187:1',
    '2:4 -> 189:0',
    '2:4-18 -> 191:2-22',
    '2:18 -> 192:0',
    '2:4 -> 194:0',
    '2:18 -> 197:1',
    '2:4 -> 199:0',
    '2:4-18 -> 203:2-36',
    '2:4-18 -> 204:2-39',
    '2:4-18 -> 205:2-32',
    '2:18 -> 206:0',
    '2:4 -> 208:0',
    '2:18 -> 210:1',
    '2:4 -> 212:0',
    '2:4-18 -> 213:2-15',
    '2:18 -> 214:0',
    '2:4 -> 216:0',
    '2:18 -> 218:1',
    '2:4 -> 220:0',
    '2:4-18 -> 221:2-18',
    '2:18 -> 222:0',
    '2:4 -> 224:0',
    '2:18 -> 226:1',
    '2:4 -> 228:0',
    '2:4-18 -> 229:2-26',
    '2:18 -> 230:0',
    '2:4 -> 232:0',
    '2:18 -> 234:1',
    '2:4 -> 236:0',
    '2:4-18 -> 238:2-14',
    '2:18 -> 239:0',
    '2:4 -> 241:0',
    '2:18 -> 244:1',
    '2:4 -> 246:0',
    '2:4-18 -> 247:2-39',
    '2:4-18 -> 248:2-30',
    '2:18 -> 249:0',
    '2:4 -> 251:0',
    '2:18 -> 253:1',
    '2:4 -> 255:0',
    '2:4-18 -> 256:2-26',
    '2:18 -> 257:0',
    '2:4 -> 259:0',
    '2:18 -> 262:1',
    '2:4 -> 264:0',
    '2:4-18 -> 265:2-36',
    '2:4-18 -> 266:2-23',
    '2:18 -> 267:0',
    '2:4 -> 269:0',
    '2:18 -> 271:1',
    '2:4 -> 273:0',
    '2:4-18 -> 274:2-20',
    '2:18 -> 275:0',
    '2:4 -> 277:0',
    '2:18 -> 279:1',
    '2:4 -> 281:0',
    '2:4-18 -> 294:2-11',
    '2:18 -> 295:0',
    '2:4 -> 297:0',
    '2:4-18 -> 298:2-11',
    '2:4-18 -> 299:2-12',
    '2:18 -> 300:0',
    '2:4 -> 302:0',
    '2:4-18 -> 303:2-12',
    '2:18 -> 304:0',
    '2:4 -> 306:0',
    '2:4-18 -> 309:2-18',
    '2:4-18 -> 310:2-11',
    '2:4-18 -> 311:2-12',
    '2:18 -> 312:0',
    '2:4 -> 314:0',
    '2:18 -> 316:1',
    '2:4 -> 317:0',
    '2:4-18 -> 318:2-12',
    '2:18 -> 319:0',
    '2:4 -> 321:0',
    '2:18 -> 323:1',
    '2:4 -> 325:0',
    '2:4-18 -> 326:2-18',
    '2:18 -> 327:0',
    '2:4 -> 329:0',
    '2:18 -> 332:1',
    '2:4 -> 334:0',
    '2:4-18 -> 336:2-20',
    '2:4-18 -> 337:2-24',
    '2:18 -> 338:0',
    '2:4 -> 340:0',
    '2:18 -> 342:1',
    '2:4 -> 344:0',
    '2:4-18 -> 346:2-17',
    '2:18 -> 347:0',
    '2:4 -> 349:0',
    '2:18 -> 351:1',
    '2:4 -> 352:0',
    '2:4-18 -> 353:2-17',
    '2:18 -> 354:0',
    '2:4 -> 356:0',
    '2:18 -> 360:1',
    '2:4 -> 362:0',
    '2:4-18 -> 370:2-24',
    '2:4-18 -> 371:2-32',
    '2:18 -> 372:0',
    '2:4 -> 374:0',
    '2:18 -> 376:1',
    '2:4 -> 378:0',
    '2:4-18 -> 380:2-17',
    '2:4-18 -> 381:2-14',
    '2:18 -> 382:0',
    '2:4-18 -> 384:0-72',
    '2:4 -> 385:0',
    '2:4-18 -> 386:2-15',
    '2:18 -> 387:0',
    '2:4 -> 389:0',
    '2:4-18 -> 390:2-26',
    '2:4-18 -> 391:2-26',
    '2:4-18 -> 392:2-21',
    '2:4-18 -> 393:2-21',
    '2:4-18 -> 394:2-16',
    '2:4-18 -> 395:2-16',
    '2:4-18 -> 396:2-16',
    '2:4-18 -> 397:2-17',
    '2:4-18 -> 398:2-17',
    '2:4-18 -> 399:2-15',
    '2:4-18 -> 400:2-15',
    '2:4-18 -> 401:2-20',
    '2:4-18 -> 402:2-40',
    '2:4-18 -> 403:2-32',
    '2:4-18 -> 404:2-31',
    '2:4-18 -> 405:2-30',
    '2:4-18 -> 406:2-17',
    '2:4-18 -> 407:2-22',
    '2:4-18 -> 408:2-24',
    '2:4-18 -> 409:2-25',
    '2:4-18 -> 410:2-26',
    '2:4-18 -> 411:2-20',
    '2:4-18 -> 412:2-29',
    '2:4-18 -> 413:2-30',
    '2:4-18 -> 414:2-40',
    '2:4-18 -> 415:2-36',
    '2:4-18 -> 416:2-29',
    '2:4-18 -> 417:2-24',
    '2:4-18 -> 418:2-32',
    '2:4-18 -> 419:2-14',
    '2:4-18 -> 420:2-20',
    '2:4-18 -> 421:2-18',
    '2:4-18 -> 422:2-19',
    '2:4-18 -> 423:2-20',
    '2:4-18 -> 424:2-16',
    '2:4-18 -> 425:2-18',
    '2:4-18 -> 426:2-15',
    '2:4-18 -> 427:2-21',
    '2:4-18 -> 428:2-23',
    '2:4-18 -> 429:2-29',
    '2:4-18 -> 430:2-27',
    '2:4-18 -> 431:2-28',
    '2:4-18 -> 432:2-29',
    '2:4-18 -> 433:2-25',
    '2:4-18 -> 434:2-26',
    '2:4-18 -> 435:2-27',
    '2:4-18 -> 436:2-24',
    '2:4-18 -> 437:2-22',
    '2:4-18 -> 438:2-24',
    '2:4-18 -> 439:2-23',
    '2:4 -> 440:2',
    '2:18 -> 441:0',
    '2:4 -> 443:0',
    '2:4-18 -> 444:2-26',
    '2:4-18 -> 445:2-26',
    '2:4-18 -> 446:2-21',
    '2:4-18 -> 447:2-21',
    '2:4-18 -> 448:2-16',
    '2:4-18 -> 449:2-16',
    '2:4-18 -> 450:2-16',
    '2:4-18 -> 451:2-17',
    '2:4-18 -> 452:2-17',
    '2:4-18 -> 453:2-15',
    '2:4-18 -> 454:2-15',
    '2:4-18 -> 455:2-20',
    '2:4-18 -> 456:2-40',
    '2:4-18 -> 457:2-32',
    '2:4-18 -> 458:2-31',
    '2:4-18 -> 459:2-30',
    '2:4-18 -> 460:2-17',
    '2:4-18 -> 461:2-22',
    '2:4-18 -> 462:2-24',
    '2:4-18 -> 463:2-25',
    '2:4-18 -> 464:2-26',
    '2:4-18 -> 465:2-20',
    '2:4-18 -> 466:2-29',
    '2:4-18 -> 467:2-30',
    '2:4-18 -> 468:2-40',
    '2:4-18 -> 469:2-36',
    '2:4-18 -> 470:2-29',
    '2:4-18 -> 471:2-24',
    '2:4-18 -> 472:2-32',
    '2:4-18 -> 473:2-14',
    '2:4-18 -> 474:2-20',
    '2:4-18 -> 475:2-18',
    '2:4-18 -> 476:2-19',
    '2:4-18 -> 477:2-20',
    '2:4-18 -> 478:2-16',
    '2:4-18 -> 479:2-18',
    '2:4-18 -> 480:2-15',
    '2:4-18 -> 481:2-21',
    '2:4-18 -> 482:2-23',
    '2:4-18 -> 483:2-29',
    '2:4-18 -> 484:2-27',
    '2:4-18 -> 485:2-28',
    '2:4-18 -> 486:2-29',
    '2:4-18 -> 487:2-25',
    '2:4-18 -> 488:2-26',
    '2:4-18 -> 489:2-27',
    '2:4-18 -> 490:2-24',
    '2:4-18 -> 491:2-22',
    '2:4-18 -> 492:2-24',
    '2:4-18 -> 493:2-23',
    '2:4 -> 494:2',
    '2:18 -> 495:0',
  ])
})

test('utilities have source maps', async () => {
  let config = {
    content: [{ raw: `text-red-500` }],
  }

  let input = css`
    @tailwind utilities;
  `

  let result = await run(input, config)
  let { sources, annotations } = parseSourceMaps(result)

  // All CSS generated by Tailwind CSS should be annotated with source maps
  // And always be able to point to the original source file
  expect(sources).not.toContain('<no source>')
  expect(sources.length).toBe(1)

  expect(annotations).toStrictEqual(['2:4 -> 1:0', '2:4-23 -> 2:4-24', '2:4 -> 3:4', '2:23 -> 4:0'])
})

test('components have source maps', async () => {
  let config = {
    content: [{ raw: `container` }],
  }

  let input = css`
    @tailwind components;
  `

  let result = await run(input, config)
  let { sources, annotations } = parseSourceMaps(result)

  // All CSS generated by Tailwind CSS should be annotated with source maps
  // And always be able to point to the original source file
  expect(sources).not.toContain('<no source>')
  expect(sources.length).toBe(1)

  expect(annotations).toEqual([
    '2:4 -> 1:0',
    '2:4 -> 2:4',
    '2:24 -> 3:0',
    '2:4 -> 4:0',
    '2:4 -> 5:4',
    '2:4 -> 6:8',
    '2:24 -> 7:4',
    '2:24 -> 8:0',
    '2:4 -> 9:0',
    '2:4 -> 10:4',
    '2:4 -> 11:8',
    '2:24 -> 12:4',
    '2:24 -> 13:0',
    '2:4 -> 14:0',
    '2:4 -> 15:4',
    '2:4 -> 16:8',
    '2:24 -> 17:4',
    '2:24 -> 18:0',
    '2:4 -> 19:0',
    '2:4 -> 20:4',
    '2:4 -> 21:8',
    '2:24 -> 22:4',
    '2:24 -> 23:0',
    '2:4 -> 24:0',
    '2:4 -> 25:4',
    '2:4 -> 26:8',
    '2:24 -> 27:4',
    '2:24 -> 28:0',
  ])
})

test('source maps for layer rules are not rewritten to point to @tailwind directives', async () => {
  let config = {
    content: [{ raw: `font-normal foo hover:foo lg:foo` }],
  }

  let utilitiesFile = postcss.parse(
    css`
      @tailwind utilities;
    `,
    { from: 'components.css', map: { prev: map } }
  )

  let mainCssFile = postcss.parse(
    css`
      @layer utilities {
        .foo {
          background-color: red;
        }
      }
    `,
    { from: 'input.css', map: { prev: map } }
  )

  // Just pretend that there's an @import in `mainCssFile` that imports the nodes from `utilitiesFile`
  let input = postcss.root({
    nodes: [...utilitiesFile.nodes, ...mainCssFile.nodes],
    source: mainCssFile.source,
  })

  let result = await run(input, config)

  let { sources, annotations } = parseSourceMaps(result)

  // All CSS generated by Tailwind CSS should be annotated with source maps
  // And always be able to point to the original source file
  expect(sources).not.toContain('<no source>')

  // And we should see that the source map for the layer rule is not rewritten
  // to point to the @tailwind directive but instead points to the original
  expect(sources.length).toBe(2)
  expect(sources).toEqual(['components.css', 'input.css'])

  expect(annotations).toEqual([
    '2:6 -> 1:0',
    '2:6 -> 2:10',
    '2:25 -> 3:0',
    '3:8 -> 4:8',
    '4:10-31 -> 5:10-31',
    '5:8 -> 6:8',
    '3:8 -> 7:8',
    '4:10-31 -> 8:10-31',
    '5:8 -> 9:8',
    '1:0 -> 10:8',
    '3:8 -> 11:8',
    '4:10-31 -> 12:10-31',
    '5:8 -> 13:8',
    '1:0 -> 14:0',
  ])
})
