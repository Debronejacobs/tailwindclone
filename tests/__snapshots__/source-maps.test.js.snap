// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apply generates source maps 1`] = `
Array [
  "2:4 -> 2:4",
  "3:6-27 -> 3:6-27",
  "4:6-33 -> 4:6-18",
  "4:6-33 -> 5:6-17",
  "4:6-33 -> 6:6-24",
  "4:6-33 -> 7:6-61",
  "5:4 -> 8:4",
  "7:4 -> 10:4",
  "8:6-39 -> 11:6-39",
  "9:6-31 -> 12:6-18",
  "9:6-31 -> 13:6-17",
  "9:6-31 -> 14:6-24",
  "9:6-31 -> 15:6-61",
  "10:4 -> 16:4",
  "13:6 -> 18:4",
  "13:6-29 -> 19:6-18",
  "13:6-29 -> 20:6-17",
  "13:6-29 -> 21:6-24",
  "13:6 -> 22:6",
  "13:29 -> 23:0",
]
`;

exports[`components have source maps 1`] = `
Array [
  "2:4 -> 1:0",
  "2:4 -> 2:4",
  "2:24 -> 3:0",
  "2:4 -> 4:0",
  "2:4 -> 5:4",
  "2:4 -> 6:8",
  "2:24 -> 7:4",
  "2:24 -> 8:0",
  "2:4 -> 9:0",
  "2:4 -> 10:4",
  "2:4 -> 11:8",
  "2:24 -> 12:4",
  "2:24 -> 13:0",
  "2:4 -> 14:0",
  "2:4 -> 15:4",
  "2:4 -> 16:8",
  "2:24 -> 17:4",
  "2:24 -> 18:0",
  "2:4 -> 19:0",
  "2:4 -> 20:4",
  "2:4 -> 21:8",
  "2:24 -> 22:4",
  "2:24 -> 23:0",
  "2:4 -> 24:0",
  "2:4 -> 25:4",
  "2:4 -> 26:8",
  "2:24 -> 27:4",
  "2:24 -> 28:0",
]
`;

exports[`preflight + base have source maps 1`] = `
Array [
  "2:4 -> 1:0",
  "2:18-4 -> 3:1-2",
  "2:18 -> 6:1",
  "2:4 -> 8:0",
  "2:4-18 -> 11:2-32",
  "2:4-18 -> 12:2-25",
  "2:4-18 -> 13:2-29",
  "2:4-18 -> 14:2-31",
  "2:18 -> 15:0",
  "2:4 -> 17:0",
  "2:4-18 -> 19:2-18",
  "2:18 -> 20:0",
  "2:4 -> 22:0",
  "2:18 -> 29:1",
  "2:4 -> 31:0",
  "2:4-18 -> 32:2-26",
  "2:4-18 -> 33:2-40",
  "2:4-18 -> 34:2-26",
  "2:4-18 -> 35:2-21",
  "2:4-18 -> 36:2-230",
  "2:4-18 -> 37:2-39",
  "2:4-18 -> 38:2-41",
  "2:18 -> 39:0",
  "2:4 -> 41:0",
  "2:18 -> 44:1",
  "2:4 -> 46:0",
  "2:4-18 -> 47:2-19",
  "2:4-18 -> 48:2-30",
  "2:18 -> 49:0",
  "2:4 -> 51:0",
  "2:18 -> 55:1",
  "2:4 -> 57:0",
  "2:4-18 -> 58:2-19",
  "2:4-18 -> 59:2-24",
  "2:4-18 -> 60:2-31",
  "2:18 -> 61:0",
  "2:4 -> 63:0",
  "2:18 -> 65:1",
  "2:4 -> 67:0",
  "2:4-18 -> 68:2-35",
  "2:18 -> 69:0",
  "2:4 -> 71:0",
  "2:18 -> 73:1",
  "2:4 -> 75:0",
  "2:4-18 -> 81:2-20",
  "2:4-18 -> 82:2-22",
  "2:18 -> 83:0",
  "2:4 -> 85:0",
  "2:18 -> 87:1",
  "2:4 -> 89:0",
  "2:4-18 -> 90:2-16",
  "2:4-18 -> 91:2-26",
  "2:18 -> 92:0",
  "2:4 -> 94:0",
  "2:18 -> 96:1",
  "2:4 -> 98:0",
  "2:4-18 -> 100:2-21",
  "2:18 -> 101:0",
  "2:4 -> 103:0",
  "2:18 -> 106:1",
  "2:4 -> 108:0",
  "2:4-18 -> 112:2-121",
  "2:4-18 -> 113:2-24",
  "2:18 -> 114:0",
  "2:4 -> 116:0",
  "2:18 -> 118:1",
  "2:4 -> 120:0",
  "2:4-18 -> 121:2-16",
  "2:18 -> 122:0",
  "2:4 -> 124:0",
  "2:18 -> 126:1",
  "2:4 -> 128:0",
  "2:4-18 -> 130:2-16",
  "2:4-18 -> 131:2-16",
  "2:4-18 -> 132:2-20",
  "2:4-18 -> 133:2-26",
  "2:18 -> 134:0",
  "2:4 -> 136:0",
  "2:4-18 -> 137:2-17",
  "2:18 -> 138:0",
  "2:4 -> 140:0",
  "2:4-18 -> 141:2-13",
  "2:18 -> 142:0",
  "2:4 -> 144:0",
  "2:18 -> 148:1",
  "2:4 -> 150:0",
  "2:4-18 -> 151:2-24",
  "2:4-18 -> 152:2-31",
  "2:4-18 -> 153:2-35",
  "2:18 -> 154:0",
  "2:4 -> 156:0",
  "2:18 -> 160:1",
  "2:4 -> 162:0",
  "2:4-18 -> 167:2-30",
  "2:4-18 -> 168:2-25",
  "2:4-18 -> 169:2-30",
  "2:4-18 -> 170:2-30",
  "2:4-18 -> 171:2-24",
  "2:4-18 -> 172:2-19",
  "2:4-18 -> 173:2-20",
  "2:18 -> 174:0",
  "2:4 -> 176:0",
  "2:18 -> 178:1",
  "2:4 -> 180:0",
  "2:4-18 -> 182:2-22",
  "2:18 -> 183:0",
  "2:4 -> 185:0",
  "2:18 -> 188:1",
  "2:4 -> 190:0",
  "2:4-18 -> 194:2-36",
  "2:4-18 -> 195:2-39",
  "2:4-18 -> 196:2-32",
  "2:18 -> 197:0",
  "2:4 -> 199:0",
  "2:18 -> 201:1",
  "2:4 -> 203:0",
  "2:4-18 -> 204:2-15",
  "2:18 -> 205:0",
  "2:4 -> 207:0",
  "2:18 -> 209:1",
  "2:4 -> 211:0",
  "2:4-18 -> 212:2-18",
  "2:18 -> 213:0",
  "2:4 -> 215:0",
  "2:18 -> 217:1",
  "2:4 -> 219:0",
  "2:4-18 -> 220:2-26",
  "2:18 -> 221:0",
  "2:4 -> 223:0",
  "2:18 -> 225:1",
  "2:4 -> 227:0",
  "2:4-18 -> 229:2-14",
  "2:18 -> 230:0",
  "2:4 -> 232:0",
  "2:18 -> 235:1",
  "2:4 -> 237:0",
  "2:4-18 -> 238:2-39",
  "2:4-18 -> 239:2-30",
  "2:18 -> 240:0",
  "2:4 -> 242:0",
  "2:18 -> 244:1",
  "2:4 -> 246:0",
  "2:4-18 -> 247:2-26",
  "2:18 -> 248:0",
  "2:4 -> 250:0",
  "2:18 -> 253:1",
  "2:4 -> 255:0",
  "2:4-18 -> 256:2-36",
  "2:4-18 -> 257:2-23",
  "2:18 -> 258:0",
  "2:4 -> 260:0",
  "2:18 -> 262:1",
  "2:4 -> 264:0",
  "2:4-18 -> 265:2-20",
  "2:18 -> 266:0",
  "2:4 -> 268:0",
  "2:18 -> 270:1",
  "2:4 -> 272:0",
  "2:4-18 -> 285:2-11",
  "2:18 -> 286:0",
  "2:4 -> 288:0",
  "2:4-18 -> 289:2-11",
  "2:4-18 -> 290:2-12",
  "2:18 -> 291:0",
  "2:4 -> 293:0",
  "2:4-18 -> 294:2-12",
  "2:18 -> 295:0",
  "2:4 -> 297:0",
  "2:4-18 -> 300:2-18",
  "2:4-18 -> 301:2-11",
  "2:4-18 -> 302:2-12",
  "2:18 -> 303:0",
  "2:4 -> 305:0",
  "2:18 -> 307:1",
  "2:4 -> 309:0",
  "2:4-18 -> 310:2-18",
  "2:18 -> 311:0",
  "2:4 -> 313:0",
  "2:18 -> 316:1",
  "2:4 -> 318:0",
  "2:4-18 -> 320:2-20",
  "2:4-18 -> 321:2-24",
  "2:18 -> 322:0",
  "2:4 -> 324:0",
  "2:18 -> 326:1",
  "2:4 -> 328:0",
  "2:4-18 -> 330:2-17",
  "2:18 -> 331:0",
  "2:4 -> 333:0",
  "2:18 -> 335:1",
  "2:4 -> 336:0",
  "2:4-18 -> 337:2-17",
  "2:18 -> 338:0",
  "2:4 -> 340:0",
  "2:18 -> 344:1",
  "2:4 -> 346:0",
  "2:4-18 -> 354:2-24",
  "2:4-18 -> 355:2-32",
  "2:18 -> 356:0",
  "2:4 -> 358:0",
  "2:18 -> 360:1",
  "2:4 -> 362:0",
  "2:4-18 -> 364:2-17",
  "2:4-18 -> 365:2-14",
  "2:18 -> 366:0",
  "2:4-18 -> 368:0-72",
  "2:4 -> 369:0",
  "2:4-18 -> 370:2-15",
  "2:18 -> 371:0",
  "2:4 -> 373:0",
  "2:4-18 -> 374:2-26",
  "2:4-18 -> 375:2-26",
  "2:4-18 -> 376:2-21",
  "2:4-18 -> 377:2-21",
  "2:4-18 -> 378:2-16",
  "2:4-18 -> 379:2-16",
  "2:4-18 -> 380:2-16",
  "2:4-18 -> 381:2-17",
  "2:4-18 -> 382:2-17",
  "2:4-18 -> 383:2-15",
  "2:4-18 -> 384:2-15",
  "2:4-18 -> 385:2-20",
  "2:4-18 -> 386:2-40",
  "2:4-18 -> 387:2-17",
  "2:4-18 -> 388:2-22",
  "2:4-18 -> 389:2-24",
  "2:4-18 -> 390:2-25",
  "2:4-18 -> 391:2-26",
  "2:4-18 -> 392:2-20",
  "2:4-18 -> 393:2-29",
  "2:4-18 -> 394:2-30",
  "2:4-18 -> 395:2-40",
  "2:4-18 -> 396:2-36",
  "2:4-18 -> 397:2-29",
  "2:4-18 -> 398:2-24",
  "2:4-18 -> 399:2-32",
  "2:4-18 -> 400:2-14",
  "2:4-18 -> 401:2-20",
  "2:4-18 -> 402:2-18",
  "2:4-18 -> 403:2-19",
  "2:4-18 -> 404:2-20",
  "2:4-18 -> 405:2-16",
  "2:4-18 -> 406:2-18",
  "2:4-18 -> 407:2-15",
  "2:4-18 -> 408:2-21",
  "2:4-18 -> 409:2-23",
  "2:4-18 -> 410:2-29",
  "2:4-18 -> 411:2-27",
  "2:4-18 -> 412:2-28",
  "2:4-18 -> 413:2-29",
  "2:4-18 -> 414:2-25",
  "2:4-18 -> 415:2-26",
  "2:4-18 -> 416:2-27",
  "2:4 -> 417:2",
  "2:18 -> 418:0",
  "2:4 -> 420:0",
  "2:4-18 -> 421:2-26",
  "2:4-18 -> 422:2-26",
  "2:4-18 -> 423:2-21",
  "2:4-18 -> 424:2-21",
  "2:4-18 -> 425:2-16",
  "2:4-18 -> 426:2-16",
  "2:4-18 -> 427:2-16",
  "2:4-18 -> 428:2-17",
  "2:4-18 -> 429:2-17",
  "2:4-18 -> 430:2-15",
  "2:4-18 -> 431:2-15",
  "2:4-18 -> 432:2-20",
  "2:4-18 -> 433:2-40",
  "2:4-18 -> 434:2-17",
  "2:4-18 -> 435:2-22",
  "2:4-18 -> 436:2-24",
  "2:4-18 -> 437:2-25",
  "2:4-18 -> 438:2-26",
  "2:4-18 -> 439:2-20",
  "2:4-18 -> 440:2-29",
  "2:4-18 -> 441:2-30",
  "2:4-18 -> 442:2-40",
  "2:4-18 -> 443:2-36",
  "2:4-18 -> 444:2-29",
  "2:4-18 -> 445:2-24",
  "2:4-18 -> 446:2-32",
  "2:4-18 -> 447:2-14",
  "2:4-18 -> 448:2-20",
  "2:4-18 -> 449:2-18",
  "2:4-18 -> 450:2-19",
  "2:4-18 -> 451:2-20",
  "2:4-18 -> 452:2-16",
  "2:4-18 -> 453:2-18",
  "2:4-18 -> 454:2-15",
  "2:4-18 -> 455:2-21",
  "2:4-18 -> 456:2-23",
  "2:4-18 -> 457:2-29",
  "2:4-18 -> 458:2-27",
  "2:4-18 -> 459:2-28",
  "2:4-18 -> 460:2-29",
  "2:4-18 -> 461:2-25",
  "2:4-18 -> 462:2-26",
  "2:4-18 -> 463:2-27",
  "2:4 -> 464:2",
  "2:18 -> 465:0",
]
`;

exports[`source maps for layer rules are not rewritten to point to @tailwind directives 1`] = `
Array [
  "2:6 -> 1:0",
  "2:6 -> 2:10",
  "2:25 -> 3:0",
  "3:8 -> 4:8",
  "4:10-31 -> 5:10-31",
  "5:8 -> 6:8",
  "3:8 -> 7:8",
  "4:10-31 -> 8:10-31",
  "5:8 -> 9:8",
]
`;
